config/routes.rb

```diff
@@ -0,0 +1,17 @@
+Rails.application.routes.draw do
+  get 'expenses/new'
+  devise_for :users
+  
+  authenticated :user do
+    root "categories#index", as: :authenticated_root
+  end
+
+  unauthenticated :user do
+    root "splash#index", as: :unauthenticated_root
+  end
+
+  resources :users
+  resources :categories, only: %i[index show new create destroy] do
+    resources :expenses, only: %i[index new create destroy]
```

Comment:
- [ ] In your expenses controller you only have 'new' and 'create' methods, please update your routes as it is for best practices.